src/process.cpp

```diff
@@ -355,6 +326,114 @@ std::string parse_env_val(bp::native_environment &env, const std::string_view &v
   return ss.str();
 }
 
+std::string validate_app_image_path(std::string app_image_path) {
+  if(app_image_path.empty()) {
+    return DEFAULT_APP_IMAGE_PATH;
+  }
+
+  // get the image extension and convert it to lowercase
+  auto image_extension = std::filesystem::path(app_image_path).extension().string();
+  boost::to_lower(image_extension);
+
+  // return the default box image if extension is not "png"
+  if(image_extension != ".png") {
+    return DEFAULT_APP_IMAGE_PATH;
+  }
+
+  // check if image is in assets directory
+  auto full_image_path = std::filesystem::path(SUNSHINE_ASSETS_DIR) / app_image_path;
+  if(std::filesystem::exists(full_image_path)) {
+    return full_image_path.string();
+  }
+  else if(app_image_path == "./assets/steam.png") {
+    // handle old default steam image definition
+    return SUNSHINE_ASSETS_DIR "/steam.png";
+  }
+
+  // check if specified image exists
+  std::error_code code;
+  if(!std::filesystem::exists(app_image_path, code)) {
+    // return default box image if image does not exist
+    BOOST_LOG(warning) << "Couldn't find app image at path ["sv << app_image_path << ']';
+    return DEFAULT_APP_IMAGE_PATH;
+  }
+
+  // image is a png, and not in assets directory
+  // return only "content-type" http header compatible image type
+  return app_image_path;
+}
+
+std::optional<std::string> calculate_sha256(const std::string &filename) {
+  EVP_MD_CTX *mdctx = EVP_MD_CTX_new();
```

Comment:
This needs to be freed. You can use the `crypto:: md_ctx_t` defined in crypto.h to handle this:
```
crypto::md_ctx_t ctx { EVP_MD_CTX_create() };
```